tests/esplora.rs

```diff
@@ -33,38 +34,36 @@ async fn test_esplora_wallet() {
     };
 
     let seed = Mnemonic::parse(MNEMONIC).unwrap().to_seed("");
-    let mut wallet = EsploraWallet::from_seed(&seed, NETWORK, ADDRESS_TYPE, esplora_url).expect("esplora_wallet");
+    let mut wallet = Wallet::from_seed(&seed, NETWORK, ADDRESS_TYPE).expect("wallet");
+    let mut blockchain_client = EsploraClient::new(esplora_url).expect("esplora_client");
 
-    wallet.full_scan(STOP_GAP, PARALLEL_REQUESTS).await.expect("full_scan");
+    let block_height = wallet.latest_checkpoint().height();
+    assert_eq!(block_height, 0);
 
-    let address0 = wallet.peek_address(KeychainKind::External, 0);
-    assert_eq!(
-        address0.address(),
-        "tb1qjtgffm20l9vu6a7gacxvpu2ej4kdcsgc26xfdz".to_string()
-    );
-
-    let balance = wallet.balance();
-    assert_eq!(balance.total().to_sat(), 0);
-
-    let address1 = wallet.next_unused_address(KeychainKind::External);
-    assert_eq!(address1.keychain(), KeychainKind::External);
-    assert_eq!(address1.index(), 0);
-
-    let address2 = wallet.reveal_next_address(KeychainKind::External);
-    assert_eq!(address2.index(), 1);
+    let full_scan_request = wallet.start_full_scan();
+    let update = blockchain_client
+        .full_scan(full_scan_request, STOP_GAP, PARALLEL_REQUESTS)
+        .await
+        .expect("full_scan");
+    wallet
+        .apply_update_at(update, Some((Date::now() / 1000.0) as u64))
```

Comment:
We could use the new helper here (if you're ok going with it)

Reply:
Done!